include/flang/Parser/dump-parse-tree.h

```diff
@@ -784,7 +784,7 @@ class ParseTreeDumper {
         std::is_same_v<T, std::int64_t> || std::is_same_v<T, std::uint64_t>) {
       ss << x;
     }
-    if (ss.tellp() != 0) {
+    if (ss.tellp() != static_cast<std::ostringstream::pos_type>(0)) {
```

Comment:
If `if (ss.tellp()) {` would work in all environments, I think it would be better.

Reply:
Done.